src/main/scala/chisel3/PolymorphicIOModule.scala

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3.experimental.{BaseModule, ExtModule, FlatIO, Param}
+
+/** A module or external module whose IO is generated from a specific generator.
+  * This module may have no additional IO created other than what is specified
+  * by its `ioGenerator` abstract member.
+  */
+sealed trait PolymorphicIOBaseModule[A <: Data] extends BaseModule {
```

Comment:
Was going to suggest an abstract class here, but I take it back, I think a trait is better.